numpy/random/_generator.pyx

```diff
@@ -3305,14 +3305,22 @@ cdef class Generator:
         --------
         Draw samples from the distribution:
 
-        >>> import numpy as np
         >>> rng = np.random.default_rng()
-        >>> s = rng.poisson(5, 10000)
+        >>> lam, size = 50, 10000
+        >>> s = rng.poisson(lam=lam, size=size)
 
-        Display histogram of the sample:
+        The mean and variance of the distribution are both ``lam``
+        
+        >>> s.mean(), s.var()
+        (50.0639, 48.63461679)  # may vary
+
+        The sample with the corresponding pmf 
 
         >>> import matplotlib.pyplot as plt
-        >>> count, bins, _ = plt.hist(s, 14, density=True)
+        >>> from scipy import stats  # doctest: +SKIP
+        >>> count, bins, _ = plt.hist(s, 30, density=True)
+        >>> x = np.arange(0, 100)
+        >>> plt.plot(x, stats.poisson.pmf(x, mu=lam))
```

Comment:
Shouldn't we use here `stem` or `scatter` plot as this is *pmf* not *pdf*?  Support of the Poisson distribution is $\mathbb{N}$ not  $\mathbb{R}$.